packages/blade/src/components/Badges/_decisions/decisions.md

```diff
@@ -0,0 +1,61 @@
+# Badge <!-- omit in toc -->
+Badges are used to show small amount of color coded metadata, which are ideal for getting users attention.
+
+<img src="./badge-thumbnail.png" width="380" />
+
+## Design
+- [Figma - Badge](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=8110%3A417)
+
+
+## API
+
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| variant | `positive`,`negative`, `notice`, `information`, `neutral` | No | `neutral` | The variant of the badge to be rendered. |
+| contrast | `low`, `high` | No | `low` | The contrast of the badge to be rendered. |
+| size | `medium`, `small` | No | `medium` | The size of the badge to be rendered. |
+| children | `string` | **Yes** | undefined | The label to be rendered within the badge. |
+| icon | `Icon` | No | undefined | The Blade `Icon` component to be rendered for the badge. |
+| labelStyle | `italic`, `normal` | No | `normal` | The font style of the badge's label. |
+| labelWeight | `bold`, `regular` | No | `regular` | The font weight of the badge's label. |
+
+### Sample Usage
+```jsx
+import { Badge } from '@razorpay/components';
+
+<Badge 
+  variant='positive'
+  contrast='high'
+  icon={InfoIcon} 
+  size='large' 
+  labelStyle='italic' 
+  labelWeight='bold'
+>
+  New
+</Badge>
+```
+
+## Open Questions
+#### Dev
+- Should the label be automatically capitalized when `labelWeight` is `bold`
+- Should we call it `labelWeight`, `labelStyle` or `fontWeight`, `fontStyle` or `textWeight`, `textStyle`?
+- Should we call it `variant` or `intent`?
```

Comment:
+1 to `variant` since that is the one used in `Alert` component. Not sure which one is used in other components. Do we use `intent` anywhere else?

Reply:
We're using a prop called `intent` that can be `positive`, `negative`, `neutral`, `information`, `notice` for `BaseButton` & `BaseLink`

- [BaseButton API](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Button/_decisions/decisions.md#basebutton-api)
- [BaseLink API](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Link/_decisions/decisions.md#baselink-api)

Reply:
cc: @divyanshu013 another possible decision that we need to take for both `Badge` & `Alert`. Calling it `variant` or `intent`?

Reply:
I think we can call it `variant` which is a more conventional prop in design libraries. `intent` could be kept for internal components and props which we aren't exposing to users.

Regardless, we should keep it consistent b/w all exported components.

Reply:
There would be a possibility that we might expose `intent` as a prop on Button to our consumers when use-cases pop up.  This means at that point, we will have `variant` & `intent` on Button where `intent` will be similar to `variant` of Alert & Badge. 

I agree variant seems more conventional for Badge & Alert but if we're treating `positive`, `negative`, `information`, etc. as _intents_ (intentions) then why not call it `intent` on Badge & Alert as well?

Reply:
We have a `variant` on button for primary, etc. So variant on some components and intent on some could also be a point of confusion. On some libraries they have separate prop for controlling the color part (which we're doing with intent).

Reply:
Let's add the points we discussed and the final call for historical purposes :)

Reply:
Added